tests/MongoDB.Driver.Tests/MongoCollectionImplTests.cs

```diff
@@ -258,6 +258,73 @@ public void Settings_should_be_set()
+        [InlineData(true, false, true)]
+        [InlineData(true, true, false)]
+        [InlineData(true, true, true)]
+        public void Aggregate_should_recognize_merge_collection_argument(bool fullForm, bool outputToDifferentDatabase, bool async)
+        {
+            var subject = CreateSubject<BsonDocument>();
+
+            CollectionNamespace expectedCollectionNamespace;
+            PipelineStageDefinition<BsonDocument, BsonDocument> stageDefinition;
+            var outputCollectionName = "outputcollection";
+            if (fullForm)
+            {
+                if (outputToDifferentDatabase)
+                {
+                    var outputDatabaseName = "outputdatabase";
+                    var outputDatabaseAndCollectionName = new BsonDocument { { "db", outputDatabaseName}, { "coll", outputCollectionName } };
+                    stageDefinition = new BsonDocument("$merge", new BsonDocument { { "into",  outputDatabaseAndCollectionName } });
+                    expectedCollectionNamespace = new CollectionNamespace(outputDatabaseName, outputCollectionName);
+                }
+                else
+                {
+                    stageDefinition = new BsonDocument("$merge", new BsonDocument { { "into", outputCollectionName } });
+                    expectedCollectionNamespace = new CollectionNamespace(subject.CollectionNamespace.DatabaseNamespace, outputCollectionName);
+                }
+            }
+            else
+            {
+                stageDefinition = new BsonDocument("$merge", outputCollectionName);
+                expectedCollectionNamespace = new CollectionNamespace(subject.CollectionNamespace.DatabaseNamespace, outputCollectionName);
+            }
+
+            var pipeline = new EmptyPipelineDefinition<BsonDocument>().AppendStage(stageDefinition);
+
+            IAsyncCursor<BsonDocument> result;
+            if (async)
+            {
+                result = subject.AggregateAsync(pipeline).GetAwaiter().GetResult();
+            }
+            else
+            {
+                result = subject.Aggregate(pipeline);
+            }
+            var aggregateCall = _operationExecutor.GetWriteCall<BsonDocument>();
+
+            var aggregateOperation = aggregateCall.Operation.Should().BeOfType<AggregateToCollectionOperation>().Subject;
+            aggregateOperation.CollectionNamespace.Should().Be(subject.CollectionNamespace);
+
+            var mockCursor = new Mock<IAsyncCursor<BsonDocument>>();
+            _operationExecutor.EnqueueResult(mockCursor.Object);
+            if (async)
+            {
+                result.MoveNextAsync().GetAwaiter().GetResult();
+            }
+            else
+            {
+                result.MoveNext();
+            }
+            var findCall = _operationExecutor.GetReadCall<IAsyncCursor<BsonDocument>>();
+
+            var findOperation = findCall.Operation.Should().BeOfType<FindOperation<BsonDocument>>().Subject;
```

Comment:
is it possible to assert `$merge` stage here (I didn't look at the code to check it)

Reply:
It's not possible for `FindOperation`, but I did it for `AggregateToCollectionOperation`.
Partially done.